exonum/src/events/noise/wrappers/sodium_wrapper/wrapper.rs

```diff
@@ -156,25 +158,30 @@ impl NoiseWrapper {
         Ok(buf)
     }
 
-    // Each message consists of the payload and 16 bytes(`TAG_LENGTH`)
-    // of AEAD authentication data. Therefore to calculate an actual message
-    // length we need to subtract `TAG_LENGTH` multiplied by messages count
-    // from `data.len()`.
-    fn decrypted_msg_len(&self, raw_message_len: usize) -> usize {
-        raw_message_len - TAG_LENGTH * (raw_message_len / MAX_MESSAGE_LENGTH)
-    }
-
-    // In case of encryption we need to add `TAG_LENGTH` multiplied by messages count to
-    // calculate actual message length.
-    fn encrypted_msg_len(&self, raw_message_len: usize) -> usize {
-        raw_message_len + TAG_LENGTH * ((raw_message_len / MAX_MESSAGE_LENGTH) + 1)
-    }
-
     fn noise_builder<'a>() -> Builder<'a> {
         Builder::with_resolver(PARAMS.parse().unwrap(), Box::new(SodiumResolver::new()))
     }
 }
 
+// Each message consists of the payload and 16 bytes(`TAG_LENGTH`)
+// of AEAD authentication data. Therefore to calculate an actual message
+// length we need to subtract `TAG_LENGTH` multiplied by messages count
+// from `data.len()`.
+fn decrypted_msg_len(raw_message_len: usize) -> usize {
+    raw_message_len - TAG_LENGTH * tag_count(raw_message_len)
+}
+
+// In case of encryption we need to add `TAG_LENGTH` multiplied by messages count to
+// calculate actual message length.
+fn encrypted_msg_len(raw_message_len: usize) -> usize {
+    raw_message_len + TAG_LENGTH * tag_count(raw_message_len)
```

Comment:
This code may be incorrect (the previous one seems to be incorrect, too). As I've pointed out in #764, the correct formula is

```rust
len + TAG_LENGTH * (floor(len / (MAX_MESSAGE_LENGTH - TAG_LENGTH)) + 1)
```

This leads to errors in tests (see the other comment).